File: owl-ui/src/config/tableConfigs/ecg.js
export const ecgConfig = {
  title: 'ECG数据',
  apiModule: 'medical',
  apiPath: 'ecg',
  permissionPrefix: 'medical:ecg',
  
  searchFields: [
    {
      prop: 'userId',
      label: '用户ID',
      type: 'input',
      props: {
        placeholder: '请输入用户ID'
      }
    },
    {
      prop: 'dataTimeRange',
      label: '数据时间范围',
      type: 'daterange',
      props: {
        'value-format': 'YYYY-MM-DD',
        'start-placeholder': '开始日期',
        'end-placeholder': '结束日期',
        'range-separator': '至',
        'clearable': true,
        'type': 'daterange'
      }
    }
  ],
  
  tableColumns: [
    {
      prop: 'userId',
      label: '用户ID',
      width: 180
    },
    {
      prop: 'dataTime',
      label: '数据时间',
      width: 180,
      type: 'time'
    },
    {
      prop: 'uploadTime',
      label: '上传时间',
      width: 180,
      type: 'time'
    },
    {
      prop: 'ecgPath',
      label: 'ECG数据路径',
      width: 300
    }
  ],
  
  formFields: [
    {
      label: '用户ID',
      prop: 'userId',
      type: 'input',
      rules: [{ required: true, message: '请输入用户ID', trigger: 'blur' }]
    },
    {
      label: '数据时间',
      prop: 'dataTime',
      type: 'datetime',
      rules: [{ required: true, message: '请选择数据时间', trigger: 'change' }]
    },
    {
      label: '记录分组ID',
      prop: 'recordGroupId',
      type: 'input'
    },
    {
      label: '上传时间',
      prop: 'uploadTime',
      type: 'datetime'
    },
    {
      label: 'ECG数据路径',
      prop: 'ecgPath',
      type: 'input',
      rules: [{ required: true, message: '请输入ECG数据路径', trigger: 'blur' }]
    },
    {
      label: '外部ID',
      prop: 'externalId',
      type: 'number'
    },
    {
      label: '元数据版本',
      prop: 'metadataVersion',
      type: 'number'
    }
  ],
  
  // 添加验证规则
  rules: {
    userId: [
      { required: true, message: '请输入用户ID', trigger: 'blur' }
    ],
    dataTime: [
      { required: true, message: '请选择数据时间', trigger: 'change' }
    ],
    ecgPath: [
      { required: true, message: '请输入ECG数据路径', trigger: 'blur' }
    ]
  },
  
  // 工具栏按钮
  toolbarButtons: [
    {
      label: '同步数据',
      type: 'primary',
      icon: 'el-icon-refresh',
      onClick: 'syncData',
      permission: 'sync'
    }
  ],
  
  // 保留buttons属性以确保兼容性
  buttons: [
    {
      label: '同步数据',
      type: 'primary',
      icon: 'el-icon-refresh',
      onClick: 'syncData',
      permission: 'sync'
    }
  ]
} 

File: owl_system/models/medical/ECG.py
from datetime import datetime
from owl_admin.ext import db

class ECG(db.Model):
    """ECG数据模型"""
    __tablename__ = 'ecg'

    id = db.Column(db.String(36), primary_key=True, comment='主键ID')
    user_id = db.Column(db.String(36), nullable=False, comment='用户ID')
    record_group_id = db.Column(db.String(36), comment='记录分组ID')
    upload_time = db.Column(db.DateTime, nullable=False, comment='上传时间')
    data_time = db.Column(db.DateTime, nullable=False, comment='数据时间')
    ecg_path = db.Column(db.String(255), nullable=False, comment='ECG数据路径')
    external_id = db.Column(db.Integer, nullable=False, comment='外部ID')
    metadata_version = db.Column(db.Integer, nullable=False, comment='元数据版本')

    def __repr__(self):
        return f'<ECG {self.id}>'

    def to_dict(self):
        """转换为字典"""
        return {
            'id': self.id,
            'userId': self.user_id,
            'recordGroupId': self.record_group_id,
            'uploadTime': self.upload_time.strftime('%Y-%m-%d %H:%M:%S') if self.upload_time else None,
            'dataTime': self.data_time.strftime('%Y-%m-%d %H:%M:%S') if self.data_time else None,
            'ecgPath': self.ecg_path,
            'externalId': self.external_id,
            'metadataVersion': self.metadata_version
        } 